api/dto/target.go

```diff
@@ -0,0 +1,187 @@
+package dto
```

Comment:
I do not think that `dto` package is a good place for this logic. In `dto` should be placed only request-response data schemes and only validation of this schemas.